uPortal-soffit-renderer/src/main/java/org/apereo/portal/soffit/renderer/SoffitRendererController.java

```diff
@@ -249,6 +282,102 @@ private String getCompletePathforParts(final String... parts) {
+                    for (int i=0; i < parameters.length; i++) {
+                        final Class<?> pType = parameterTypes[i];
+                        // At present, these are the parameter types we support...
+                        if (HttpServletRequest.class.equals(pType)) {
+                            parameters[i] = req;
+                        } else if (HttpServletResponse.class.equals(pType)) {
+                            parameters[i] = res;
+                        } else if (PortalRequest.class.equals(pType)) {
+                            parameters[i] = portalRequest;
+                        } else if (Bearer.class.equals(pType)) {
+                            parameters[i] = bearer;
+                        } else if (Preferences.class.equals(pType)) {
+                            parameters[i] = preferences;
+                        } else if (Definition.class.equals(pType)) {
+                            parameters[i] = definition;
+                        } else {
+                            final String msg = "Unsupported parameter type for SoffitModelAttribute method:  " + pType;
+                            throw new UnsupportedOperationException(msg);
+                        }
+                    }
+                    try {
+                        final Object value = m.invoke(bean, parameters);
+                        rslt.put(modelAttributeName, value);
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        final String msg = "Failed to evaluate the specified model attribute:  " + sma.value();
+                        throw new RuntimeException(msg);
+                    }
+                } else {
+                    final String msg = "Unsuppored AnnotatedElement type:  " + AnnotatedElement.class.getName();
+                    throw new UnsupportedOperationException(msg);
+                }
+            }
+        }
+
+        logger.debug("Calculated the following model attributes for viewName='{}':  {}", viewName, rslt);
+
+        return rslt;
+
+    }
+
+    private boolean attributeAppliesToView(SoffitModelAttribute attributeAnnotation, String viewName) {
+        final Pattern pattern = Pattern.compile(attributeAnnotation.viewRegex());
+        final Matcher matcher = pattern.matcher(viewName);
+        return matcher.matches();
+    }
+
+    private void configureCacheHeaders(final HttpServletResponse res, final String module) {
+
+        final String cacheScopeProperty = String.format(CACHE_SCOPE_PROPERTY_FORMAT, module);
+        final String cacheScopeValue = environment.getProperty(cacheScopeProperty);
+        logger.debug("Selecting cacheScopeValue='{}' for property '{}'", cacheScopeValue, cacheScopeProperty);
+
+        final String cacheMaxAgeProperty = String.format(CACHE_MAXAGE_PROPERTY_FORMAT, module);
+        final String cacheMaxAgeValue = environment.getProperty(cacheMaxAgeProperty);
```

Comment:
I usually include units of measure in variable names.  I probably didn't in this case because the variable represents a standard HTTP header (with a name that doesn't include a unit).  We could nevertheless add it.